src/System.Management.Automation/engine/hostifaces/MshHostUserInterface.cs

```diff
@@ -1101,19 +1086,22 @@ internal string Path
         /// </summary>
         internal bool IncludeInvocationHeader { get; set; }
 
-        /// <summary>
-        /// The encoding of this transcript, so that appending to it
-        /// can be done correctly.
-        /// </summary>
-        internal Encoding Encoding { get; private set; }
-
         /// <summary>
         /// Logs buffered content to disk. We use this instead of File.AppendAllLines
         /// so that we don't need to pay seek penalties all the time, and so that we
         /// don't need append permission to our own files.
         /// </summary>
         internal void FlushContentToDisk()
         {
+            Encoding GetPathEncoding()
+            {
+                using (StreamReader reader = new StreamReader(this.Path, Utils.utf8NoBom, detectEncodingFromByteOrderMarks: true))
+                {
+                    _ = reader.Read();
+                    return reader.CurrentEncoding;
+                }
+            }
```

Comment:
Recently we activated a Roslyn analyzer which suggest to make this static if possible. Other suggest simplify `using`.
```suggestion
            static Encoding GetPathEncoding(string path)
            {
                using StreamReader reader = new StreamReader(path, Utils.utf8NoBom, detectEncodingFromByteOrderMarks: true);
                _ = reader.Read();
                return reader.CurrentEncoding;
            }
```